### ISSUE TITLE
Color/disabled mixins removed properties from the api docs

### ISSUE THREAD

**Issue by willshowell** [angular/material2#5284]
#### Bug, feature request, or proposal:
Docs bug

#### What is the expected behavior?
A component that implements CanDisable or CanColor should still show `color` and `disabled` as valid properties with descriptions in the component's api reference.

#### Reproduction
Checkbox implements both, but has no reference: https://material.angular.io/components/checkbox/api

**Comment by DevVersion** [comment]
Yeah this is a known issue. We need to see if we can somehow get a Dgeni processor that properly handles mixins

**Comment by jelbourn** [comment]
@DevVersion we could alternatively try to pull the properties from the interface (e.g. `CanDisable`)

**Comment by willshowell** [comment]
@DevVersion have you had the chance to look into this? Looking at the [dgeni-packages CHANGELOG](https://github.com/angular/dgeni-packages/blob/master/CHANGELOG.md), it looks like there have been some improvements to how `implements` and `extends` work, but I'm not familiar enough with Dgeni to know if they're valuable.

**Comment by DevVersion** [comment]
@willshowell Yeah that's the plan. Just wanted to upgrade everything and revisit this afterwards.
